addons/point_of_sale/models/pos_payment.py

```diff
@@ -0,0 +1,26 @@
+from odoo import api, fields, models
+
+
+class PosPayment(models.Model):
+    """ This model will be used to prevent using account.payment for each pos.order.
+        The pos.payment for each pos.session will be combined to generate one (or more if multiple types)
+        account.payment. This will reduce the number of account.payment, account.move and account.move.line
+        that will be created when closing a pos session.
+    """
+
+    _name = "pos.payment"
+    _description = "Point of Sale Payments"
+    _order = "id desc"
+
+    name = fields.Char(readonly=True, copy=False)
+    pos_order_id = fields.Many2one(comodel_name='pos.order', string='Pos Order')
+    amount = fields.Monetary(string='Payment Amount', required=True, readonly=True, help="Total amount of the payment.")
+    # TODO jcb: company currency is not fully correct. Need to confirm this with PO.
+    currency_id = fields.Many2one('res.currency', string='Currency', required=True, readonly=True, default=lambda self: self.env.company.currency_id)
```

Comment:
you cannot use self.env.company.currency_id. probably we cannot really set a default as it will depend either, of the currency_id on pos_order/pos_config